src/app/panels/image-panel/image-panel.component.html

```diff
@@ -1,21 +1,33 @@
-<evt-panel [showSecondaryContent]="msDescOpen"
-    [hideHeader]="false"
-    [hideFooter]="true">
+<evt-panel [showSecondaryContent]="isSecondaryContentOpened()">
     <div header-left>
-        <!-- TODO: Add dropdowns for navigation -->
-        <evt-ms-desc-selector #msDesc (selectionChange)="setMsDescID($event)" (msDescOpen)="setMsDescOpen($event)"></evt-ms-desc-selector>
+    <ng-select 
+        class="mr-1 imgSelect"
+        bindLabel="label"
+        bindValue="label"
+        [searchable]="true" 
+        [clearable]="false"
+        [ngModel]="currentImg"
+        (change)="toggleImg($event)">
+        <ng-option [value]="item" *ngFor="let item of images;">
+            {{ item.label || item.url.split('/').pop().split('.')[0] }}
+        </ng-option>
+    </ng-select>
+    <evt-ms-desc-selector (selectionChange)="setMsDescID($event)" (msDescOpen)="isMsDescOpen($event)"></evt-ms-desc-selector>
     </div>
     <div content>
         <evt-osd
             *ngIf="viewerData"
-            [viewerData]="viewerData">
+            [viewerData]="viewerData"
+            [page]="imageIndex"
+            (pageChange)="eventPageImg($event)">
         </evt-osd>
         <p *ngIf="!viewerData">Found no source file</p>
     </div>
-
     <div secondary-content>
-        <div *ngIf="msDescOpen">
-            <evt-ms-desc [data]="currentMsDesc$ | async"></evt-ms-desc>    
+        <div *ngIf="(msDesc$ | async) as msDescs">
+            <div *ngFor="let msDesc of msDescs">
```

Comment:
You can use the async binding directly in the loop:
```
<div *ngFor="let msDesc of msDescs$ | async">
```

FYI, if you need to add a level to handle logic (like you did here), but don't need additional HTML you can use `ng-container`.